lua/nvim-tree/renderer/decorator/size.lua

```diff
@@ -0,0 +1,134 @@
+local HL_POSITION = require("nvim-tree.enum").HL_POSITION
+local ICON_PLACEMENT = require("nvim-tree.enum").ICON_PLACEMENT
+local view = require "nvim-tree.view"
+
+local Decorator = require "nvim-tree.renderer.decorator"
+
+---@class DecoratorSize: Decorator
+---@field width_cutoff integer
+---@field noshow_folder_size_glyph string
+---@field column_width integer
+---@field show_folder_size boolean
+---@field units string[]
+---@field format_unit function
+---@field format_size function
+local DecoratorSize = Decorator:new()
+
+---@param opts table
+---@return DecoratorSize
+function DecoratorSize:new(opts)
+  local o = Decorator.new(self, {
+    enabled = opts.renderer.size.enable,
+    hl_pos = HL_POSITION.none,
+    icon_placement = ICON_PLACEMENT.right_align,
+  })
+  ---@cast o DecoratorSize
+
+  if not o.enabled then
+    return o
+  end
+
+  o.width_cutoff = opts.renderer.size.width_cutoff
+  o.noshow_folder_size_glyph = opts.renderer.size.noshow_folder_size_glyph or ""
+  o.column_width = opts.renderer.size.column_width
+  o.show_folder_size = opts.renderer.size.show_folder_size
+  o.units = { "B", "KB", "MB", "GB", "TB", "PB", "EB", "ZB", "YB" }
+
+  -- Guaranteed to be a function from previous setup
+  o.format_unit = opts.renderer.size.format_unit
+  o.format_size = function(size)
```

Comment:
Could this be a method? It doesn't look like there's anything specific in there.

Reply:
I considered allowing format_size to be user configurable, but gave up on that.

Reply:
If users want it we can add it.